intelmq/bots/experts/modify/expert.py

```diff
@@ -19,60 +33,73 @@ def matches(self, identifiers, event, *rules):
         for rule in rules:
             condition.update(rule)
 
+        matches = {}
+
         for name, rule in condition.items():
             # empty string means non-existant field
             if rule == '':
                 if name in event:
-                    return False
+                    return None
                 else:
                     continue
             if name not in event:
-                return False
+                return None
             if not isinstance(rule, type(event[name])):
                 if isinstance(rule, str) and isinstance(event[name], (int, float)):
-                    if not re.search(rule, str(event[name])):
-                        return False
+                    match = re.search(rule, str(event[name]))
+                    if match is None:
+                        return None
+                    else:
+                        matches[name] = match
                 else:
                     self.logger.warn("Type of rule ({!r}) and data ({!r}) do not "
                                      "match in {!s}, {}!".format(type(rule), type(event[name]), identifiers, name))
             elif not isinstance(event[name], str):  # int, float, etc
                 if event[name] != rule:
-                    return False
+                    return None
             else:
-                if not re.search(rule, event[name]):
-                    return False
+                match = re.search(rule, event[name])
+                if match is None:
+                    return None
+                else:
+                    matches[name] = match
 
-        return True
+        return matches
 
-    def apply_action(self, event, action):
+    def apply_action(self, event, action, matches={}):
```

Comment:
What's the purpose of the default value for the matches parameter? It's not needed. All places where the method is called provide a value for this parameter. Also, the default value is mutable, which can lead to subtle problems ([see Python FAQ](https://docs.python.org/3/faq/programming.html#why-are-default-values-shared-between-objects))